napari/_qt/_tests/test_app.py

```diff
@@ -33,6 +33,10 @@ def get_app_id():
 
 def test_run_outside_ipython(qapp, monkeypatch):
     """Test that we don't incorrectly give ipython the event loop."""
+    from IPython.core.interactiveshell import InteractiveShell
+
+    InteractiveShell._instance = None
```

Comment:
```suggestion
    assert hasattr(InteractiveShell, "_instance ")
    InteractiveShell._instance = None
```

to detect removes of `_instance` attribute?

Reply:
no need, the metaclass ensure `_instance` always exists. 

```
>>> from IPython.core.interactiveshell import InteractiveShell
>>> del InteractiveShell._instance
>>> InteractiveShell._instance is None
True
```

Reply:
I was thinking about future ipython releases. I know that it is not planned to delete this attribute, but it may happen because of some new requirements.

Reply:
That is fairly unlikely, InteractiveShell is from Traitlets' SingletonConfigurable. If that ever happens it's going to be several years in the making.

I think that's a fairly weird thing we are doing already, so I'm more tempted to fail if `_instance` does not exists, because if it does not then it means IPython has changed enough that it's worth revisiting this fixture..